### PR TITLE
Use `Homebrew::CLI::Parser`

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
This helps us with consistent help output, parsing invalid options and accessing options without using ARGV directly.

While we're here, use this chance to simplify the code around argument parsing and different commands.

CC @bfontaine for review.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Tests are now 💚

File: cmd/alias.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    if Homebrew.args.edit?
+      if arg.blank?
+        Aliases.edit_all
+      else
+        Aliases.edit split_arg
+      end
+    elsif /.=./.match?(arg)
+      Aliases.add split_arg
+    elsif arg.present?
+      Alises.show arg
```
Comment (bfontaine) [inline cmd/alias.rb]: typo: `Aliases`

File: cmd/alias.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    Aliases.init
+
+    if Homebrew.args.edit?
+      if arg.blank?
+        Aliases.edit_all
+      else
+        Aliases.edit split_arg
+      end
+    elsif /.=./.match?(arg)
+      Aliases.add split_arg
```
Comment (bfontaine) [inline cmd/alias.rb]: `*split_arg` otherwise it fails because `add` takes two arguments.

File: cmd/alias.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    arg = Homebrew.args.named.first
+    split_arg = arg.split("=", 2) if arg.present?
+
+    Aliases.init
+
+    if Homebrew.args.edit?
+      if arg.blank?
+        Aliases.edit_all
+      else
+        Aliases.edit split_arg
```
Comment (bfontaine) [inline cmd/alias.rb]: Should be `arg` here.

File: lib/alias.rb [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          <<~EOS
+            #
+            # This is a Homebrew alias script. It'll be called when the
+            # user type `brew #{name}`. Any remaining argument is
+            # passed to this script. You can retrieve those with $*, or
+            # the first one only with $1. Please keep your script on
+            # one line.
+
+            # TODO Replace the line below with your script
+            echo 'Hello I'm brew alias "#{name}" and my args are:' $1
```
Comment (bfontaine) [inline lib/alias.rb]: Note related to your PR but I’m just now realizing that this command doesn’t work due to the `'` inside the string. It should be `I’m`, not `I'm`.

**Comment by MikeMcQuaid** [comment]
Thanks for the review! All comments addressed.

**Comment by bfontaine** [comment]
Thanks Mike!

**Pull request merged**